executor/union_scan.go

```diff
@@ -108,6 +108,31 @@ type UnionScanExec struct {
 
 // Next implements Execution Next interface.
 func (us *UnionScanExec) Next(goCtx goctx.Context) (Row, error) {
+	row, err := us.getOneRow(goCtx)
+	return row, errors.Trace(err)
+}
+
+// NextChunk implements the Executor NextChunk interface.
+func (us *UnionScanExec) NextChunk(goCtx goctx.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	for i, batchSize := 0, us.ctx.GetSessionVars().MaxChunkSize; i < batchSize; i++ {
+		row, err := us.getOneRow(goCtx)
```

Comment:
`getOneRow` will call `getSnapshotRow`, `getSnapshotRow` will call `Next` method of child.